spark/src/main/scala/org/apache/spark/sql/comet/DecimalPrecision.scala

```diff
@@ -107,11 +108,4 @@ object DecimalPrecision {
       case e => e
     }
   }
-
-  object DecimalExpression {
-    def unapply(e: Expression): Option[(Int, Int)] = e.dataType match {
-      case t: DecimalType => Some((t.precision, t.scale))
-      case _ => None
-    }
-  }
```

Comment:
Hmm, doesn't this work for Spark 4.0?

Reply:
Spark 4.0 has `org.apache.spark.sql.types.DecimalExpression` So I moved this object to shims

Reply:
But I think this `DecimalExpression.unapply` also works for Spark 4.0 right? We don't need to use `DecimalExpression` if current approach works both for Spark 3.x and Spark 4.0.

Reply:
Yes, it works. I think this code was copied from Spark 3.5+. This code should have been in shims to start with.

When we drop Spark 3.x support in the future, we no longer need this object. We can just directly use `org.apache.spark.sql.types.DecimalExpression`

If keeping the current copy of code is preferred, I can put it back though

Reply:
Restored the original